### DISCUSSION TITLE
Is the smoothing setting with Dexcom G6 the same as in the Dexcom app?

### DISCUSSION THREAD

**Discussion by dar-kod** [NightscoutFoundation/xDrip#3030] [General]
I currently use the Dexcom G6. I would like to check if the smoothing setting is correct. The original Dexcom application has filtered data. Is it possible to get the same smoothing as in the Dexcom application, or is there a better smoothing algorithm?

**Comment by Navid200** [comment]
xDrip does not do any smoothing to the G6 data.
xDrip has a graph smoothing.  But, that only smooths out what is shown on the screen without modifying the data points.

Why do you think that the Dexcom app smooths the data?  I'm not saying it doesn't.  I would just like to learn more.

**Comment by dar-kod** [comment]
> Why do you think that the Dexcom app smooths the data? I'm not saying it doesn't. I would just like to learn more.

Thanks for the explanation. If you still have AAPS, you will see **Smoothing** (select smoothing algorithm) in **Config Builder**. See attached screenshot. I do not know if it is data smoothing or graph smoothing. I need to know which smoothing algorithm to select in relation to xDrip as the source of BG. Smoothing is very important, otherwise AAPS may give wrong treatment.

![1692535912873](https://github.com/NightscoutFoundation/xDrip/assets/24446806/546185d3-f1f0-4775-9e50-519cf369c15e)

**Reply by Navid200** [reply]
xDrip does not smooth G6 data.  It has an option that smooths the graph only.

@jamorham can correct me if I am wrong.

You may want to ask the AAPS group how you should use their app.

**Reply by dar-kod** [reply]
Yes, thank you.

**Reply by dar-kod** [reply]
For those using AAPS and xDrip+, "Average smoothing" seems to be a good choice. For the G7, "Exponential smoothing" is mandatory.

[Source](https://androidaps.readthedocs.io/en/latest/Hardware/DexcomG7.html#build-a-new-signed-apk-from-the-dev-branch):

The smoothing of glucose values can be activated by enabling the “Average smoothing” or “Exponential Smoothing” plugin in the Config Builder. To disable select the “No Smoothing” option. “Exponential smoothing” is more aggressive and rewrites the newest Glucose Value but is good in dealing with heavy noise. “Average smoothing” is much like the back smoothing that was in BYODA G6 and only rewrites the past values but not the current value and therefore has a faster response time.

Exponential Smoothing MUST be enabled for meaningful use of the G7 values.
